### ISSUE TITLE
Unhandled NullPointerException in BayesianHetPulldownCalculator

### ISSUE THREAD

**Issue by dpmccabe** [broadinstitute/gatk-protected#720]
I ran `GetBayesianHetCoverage` on some normal ICE exomes today and got a NullPointerException on one of the samples.

**Call**
```
java -Xmx9g -Djava.library.path=/broad/software/free/Linux/redhat_6_x86_64/pkgs/hdfview_2.9/HDFView/lib/linux/ \
  -jar /xchip/scarter/dmccabe/gatk-protected/build/libs/gatk-protected.jar \
  --reference /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta \
  --normal /xchip/bloodbiopsy/data/cell_free_DNA/18Dec15SR_cfDNA_WES/mergedBamFiles/FC15884027.markDuplicates.bam \
  --snpIntervals /xchip/cga/reference/hg19/whole_exome_illumina_coding_v1_plus_10bp_padding_minus_mito.Homo_sapiens_assembly19.targets.interval_list \
  --normalHets /xchip/scarter/dmccabe/lb_germline/out/HD5.ctDNA.EX/HD5.ctDNA.EX.hets \
  --help false --version false --verbosity DEBUG --QUIET false  \
  --readDepthThreshold 15 --minimumMappingQuality 30 --minimumBaseQuality 20 --VALIDATION_STRINGENCY SILENT \
  --hetCallingStringency 30.0 --minimumAbnormalFraction 0.5 --maximumAbnormalFraction 0.8 \
  --maximumCopyNumber 2 --quadratureOrder 200 --errorAdjustmentFactor 1.0 --use_jdk_deflater false
```

**Exception**
```
java.lang.NullPointerException
	at org.broadinstitute.hellbender.tools.exome.pulldown.BayesianHetPulldownCalculator.getHetLogLikelihood(BayesianHetPulldownCalculator.java:174)
	at org.broadinstitute.hellbender.tools.exome.pulldown.BayesianHetPulldownCalculator.getHetPulldown(BayesianHetPulldownCalculator.java:343)
	at org.broadinstitute.hellbender.tools.exome.GetBayesianHetCoverage.runNormalOnly(GetBayesianHetCoverage.java:223)
	at org.broadinstitute.hellbender.tools.exome.GetBayesianHetCoverage.doWork(GetBayesianHetCoverage.java:306)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:108)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:166)
	at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:185)
	at org.broadinstitute.hellbender.Main.instanceMain(Main.java:76)
	at org.broadinstitute.hellbender.Main.main(Main.java:92)
```

The full log is available at [this Gist](https://gist.github.com/dpmccabe/96f4818aa9e3b9388b627c5ff5ea48c1). I'm at commit [f8ead0d](https://github.com/broadinstitute/gatk-protected/commit/f8ead0d) of this repo and the exception occurred [at this line](https://github.com/broadinstitute/gatk-protected/blob/f8ead0d8a1273f1b416dbf249741300dc84c8879/src/main/java/org/broadinstitute/hellbender/tools/exome/pulldown/BayesianHetPulldownCalculator.java).

The other nine normals in my set were processed successfully, as was this particular sample when I switched to `GetHetCoverage`, which seems to iterate over the SNPs differently. `GetBayesianHetCoverage` worked on this sample when I swapped out the SNP list to a more inclusive one.

It would be helpful if this exception were handled in a way that would let me identify the problematic SNP.

**Comment by mbabadi** [comment]
Hi Devin, thank you for reporting this. I am almost confident that the cause of the NullPointerException is a non-ACTG (e.g. "N") base in the reference genome at one of the provided SNP intervals, possibly in the Ychr. The reference you have used has a number of "N" bases toward the end of the Y contig. GetHetCoverage does not run into a similar problem because it uses a simpler test for detecting Hets that only requires the reference base counts at each pileup. In any event, I will make sure non-ACTG reads are filtered out.

**Comment by mbabadi** [comment]
This should fix the problem:
https://github.com/broadinstitute/gatk-protected/pull/722

**Comment by mbabadi** [comment]
Verified, #722 resolves the issue. By the way, I could also log a debug-level message about bad SNP sites (i.e. if the reference genome is nonsense at that site) -- would that be of any use to you?

**Comment by dpmccabe** [comment]
Yes, a log message would definitely be helpful if it can be added easily. I was dropping the sex chromosomes downstream in this instance, but this might not always be the case. Thanks!

**Comment by mbabadi** [comment]
addressed in #722, closing.
